Other structure for Data Encryption Service

andrzejsamson opened the pull request:
Changed the structure of Data Encryption Service. Removed redundant Model class and separated GUI from logic.

andrzejsamson committed:
Other structure for Data Encryption Service

Changed the structure of Data Encryption Service. Removed redundant Model class and separated GUI from logic.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/devonfw-forge/mrchecker-source?pullRequest=52) <br/>All committers have signed the CLA.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionController.java:
```diff
@@ -0,0 +1,14 @@
+	
+	void onDecrypt(CryptParams cryptParams);
+	
+	void onKey(JTextField keyTextField);
```
This should not be a part of DataEncryptionController

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java:
```diff
@@ -0,0 +1,78 @@
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IView;
+
+public class DataEncryptionControllerImp implements DataEncryptionController {
+	private final IView view;
```
Change IView to DataEncryptionView

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java:
```diff
@@ -0,0 +1,78 @@
+			view.setEncryptionFieldValue(DataEncryptionService.getInstance()
+					.encrypt(cryptParams.getText()));
+		} catch (BFSecureModuleException e) {
+			showMessageDialog(null, "Could not encrypt because:\n" + e.getMessage());
```
Do not show dialog here. Throw unchecked exception and handle it in the view.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java:
```diff
@@ -0,0 +1,78 @@
+	}
+	
+	@Override
+	public void onDecrypt(CryptParams cryptParams) {
```
onEncrypt and onDecrypt have duplicates. Please refactor. You can use functional interface to implement that.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java:
```diff
@@ -0,0 +1,78 @@
+	}
+	
+	@Override
+	public void onKey(JTextField keyTextField) {
```
Let the view control opening files.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/DataEncryptionGUI.java:
```diff
@@ -0,0 +1,67 @@
+	private JLabel decryptionResultLabel;
+	
+	public DataEncryptionGUI() {
+		dataEncryptionController = new DataEncryptionControllerImp(this);
```
This cannot be implemented like that. We need another way to instantiate and pass DataEncryptionController implementation.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/view/IView.java:
```diff
@@ -0,0 +1,11 @@
+	
+	void setDecryptionFieldValue(String text);
+	
+	void setKeyFieldValue(String text, JTextField keyTextField);
```
Let's talk about that

andrzejsamson committed:
Changes in class schema

Schema of classes changed, so now there are more able to test. Business logic separated from GUI form and initialization. GUI code generated for form so it can be run from jar file. Build added to pom file to declare the manifest of the jar file.

andrzejsamson commented:
TODO:
-tests for business logic from app
-instruction how to run and use the app

andrzejsamson committed:
Refactor of interface names

Interface names refactored so they are have the same convention with the project

other019 committed:
use shade plugin to prepare small uber jar

andrzejsamson committed:
Merge pull request #53 from devonfw-forge/feature/50.1-remove-redundant-model-pack-to-jar

use shade plugin to prepare small uber jar

andrzejsamson committed:
Tests for DataEncryptionApp added, Logger from Exceptions removed, pom modified

Tests for DataEncryptionController added. Tests for DataEncryptionGUI added using AssertJ Swing library. BfLogger from BFSecureModuleException removed. Pom file modified so now the DataEncryptionApp does not need to have log4j included

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+@IntegrationTest
+@ResourceLock(value = "SingleThread")
+public class DataEncryptionGUITest {
+	private static final IDataEncryptionController dataEncryptionController = new DataEncryptionController();
```
Using a concrete class. A mock should be used.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, startsWith("ENC("));
```
"ENC(" is coupled with DataEncrytpionController. Use mock instead and define return value.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+				.enterText(value);
+		dataEncryptionGUIFixture.button("chooseEncryptionKeyButton")
+				.click();
+		dataEncryptionGUIFixture.fileChooser()
```
Do not use file. Give some explicit value.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+	}
+	
+	@Test
+	public void shouldShowMessageDialogWhenSecretIsEmptyForEncryption() {
+		CryptParams cryptParams = new CryptParams("", value);
+		dataEncryptionGUIFixture.textBox("encryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("encryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("encryptButton")
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
+	}
+	
+	@Test
+	public void shouldShowMessageDialogWhenSecretIsTooShortForEncryption() {
+		CryptParams cryptParams = new CryptParams("123", value);
+		dataEncryptionGUIFixture.textBox("encryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("encryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("encryptButton")
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
```
Code duplication?

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+				.enterText(cipherForSecretDataFile);
+		dataEncryptionGUIFixture.button("chooseDecryptionKeyButton")
+				.click();
+		dataEncryptionGUIFixture.fileChooser()
+				.setCurrentDirectory(new File(System.getProperty("user.dir")))
+				.selectFile(new File(System.getProperty("user.dir") + "/src/resources/secretData"))
+				.approve();
+		dataEncryptionGUIFixture.button("decryptButton")
```
Do not use a file.

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+	@Test
+	public void shouldShowMessageDialogWhenSecretIsTooShortForDecryption() {
+		dataEncryptionGUIFixture.tabbedPane()
+				.selectTab("Decryption");
+		CryptParams cryptParams = new CryptParams("123", cipher);
+		dataEncryptionGUIFixture.textBox("decryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("decryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("decryptButton")
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("decryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
+	}
+	
+	@Test
+	public void shouldShowMessageDialogWhenValueIsNotEncryptedForDecryption() {
+		dataEncryptionGUIFixture.tabbedPane()
+				.selectTab("Decryption");
+		CryptParams cryptParams = new CryptParams(secret, value);
+		dataEncryptionGUIFixture.textBox("decryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("decryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("decryptButton")
+				.click();
+		dataEncryptionGUIFixture.dialog()
+				.button()
+				.click();
+		
+		String result = dataEncryptionGUIFixture.textBox("decryptionResultTextField")
+				.text();
+		
+		assertThat(result, is(equalTo("")));
```
Code duplication?

michalbabiarz commented on mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java:
```diff
@@ -0,0 +1,325 @@
+import com.capgemini.mrchecker.test.core.utils.encryption.view.DataEncryptionGUI;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IDataEncryptionView;
+
+@IntegrationTest
```
Can be a UnitTest

andrzejsamson committed:
Supplier added to GUI for easier test purposes. Unit tests changed to avoid code duplication and use of file chooser.
